lib/types/index.d.ts

```diff
@@ -1171,6 +1171,14 @@ export namespace Rule {
 		 * Mandatory for rules that provide suggestions.
 		 */
 		hasSuggestions?: boolean | undefined;
+		/**
+		 * Specifies the dialects the rule supports.
+		 */
+		dialects?: ("javascript" | "typescript")[] | undefined;
+		/**
+		 * Specifies the language the rule supports.
+		 */
+		language?: "javascript" | "typescript" | undefined;
```

Comment:
These fields are already specified in `RuleDefinition` generically, so we don't need to add them here.

Reply:
```suggestion
```